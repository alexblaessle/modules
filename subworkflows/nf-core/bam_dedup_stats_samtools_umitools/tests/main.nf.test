nextflow_workflow {

    name "Test Workflow BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS"
    script "../main.nf"
    workflow "BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS"
    tag "samtools"
    tag "samtools/index"
    tag "samtools/stats"
    tag "samtools/idxstats"
    tag "samtools/flagstat"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "bam_dedup_stats_samtools_umitools"
    tag "bam_stats_samtools"
    tag "umitools"
    tag "umitools/dedup"

    test("test_bam_dedup_stats_samtools_umitools") {

        when {
            params {
                outdir   = "$outputDir"
            }
            workflow {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
                ]
                input[1] = false
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert snapshot(workflow.out).match()}
            )
        }

    }

}
