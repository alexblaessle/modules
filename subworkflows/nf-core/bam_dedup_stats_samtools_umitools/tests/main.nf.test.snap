{
    "test_bam_dedup_stats_samtools_umitools": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,350e942a0d45e8356fa24bc8c47dc1ed"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam.bai:md5,2cdd0ec4c4877d497c68edc54aa6cb5e"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats:md5,d4d6860116eb212e4aad30082ed43f83"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test.flagstat:md5,0bb716e40fae381b97484b58e0b16efe"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test.idxstats:md5,1adb27b52d4d64b826f48b59d61dcd4d"
                    ]
                ],
                "6": [
                    "versions.yml:md5,2be03e07f6358cfdd59e31b803b7a72a",
                    "versions.yml:md5,42d983b7367b2b868181b4df009bf0ad",
                    "versions.yml:md5,d75ce9fe6c440dc09565c5aea6770a06",
                    "versions.yml:md5,fde22379558d4561bd12e4a4377b46e4",
                    "versions.yml:md5,ffa833166793450237feb32e0d3786f6"
                ],
                "bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam.bai:md5,2cdd0ec4c4877d497c68edc54aa6cb5e"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,350e942a0d45e8356fa24bc8c47dc1ed"
                    ]
                ],
                "csi": [
                    
                ],
                "flagstat": [
                    [
                        {
                            "id": "test"
                        },
                        "test.flagstat:md5,0bb716e40fae381b97484b58e0b16efe"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.idxstats:md5,1adb27b52d4d64b826f48b59d61dcd4d"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats:md5,d4d6860116eb212e4aad30082ed43f83"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2be03e07f6358cfdd59e31b803b7a72a",
                    "versions.yml:md5,42d983b7367b2b868181b4df009bf0ad",
                    "versions.yml:md5,d75ce9fe6c440dc09565c5aea6770a06",
                    "versions.yml:md5,fde22379558d4561bd12e4a4377b46e4",
                    "versions.yml:md5,ffa833166793450237feb32e0d3786f6"
                ]
            }
        ],
        "timestamp": "2023-10-18T10:00:33.895257718"
    }
}